### ISSUE TITLE
hooks not being cleared per scenario

### ISSUE THREAD

**Issue by darrinholst** [cucumber-attic/cuke4duke#54]
Hooks are being registered once per scenario, but they are never being cleared so you end up with them being called multiple times per scenario. I modified HookSteps.java to show this here:

http://github.com/darrinholst/cuke4duke/commit/18884e20e4b8d018951a46129d5718916f77b596

which gets you failures like:

  Feature: Call steps
    In order to reuse my steps
    I, the programmer want to call them from a step definition by name

```
Scenario: Call a step      # features/call_step.feature:5
  When I call another step # CallingSteps.iCallAnotherStep()
  Then magic should happen # CalledSteps.magicShouldHappen()
```

  Feature: Simple

```
Scenario: 3 green cukes            # features/demo.feature:3
  Given I have 3 green cukes       # StuffSteps.iHaveNCukes(int,String)
  When I add a table               # StuffSteps.iAddATable(Table)
    | a | b |
    | 1 | 2 |
  Then I should have 3 green cukes # StuffSteps.iShouldHaveNCukes(int,String)
  And I should have 0 yellow cukes # StuffSteps.iShouldHaveNCukes(int,String)
    java.lang.AssertionError: expected:<1> but was:<2> (NativeException)

Scenario: 4 green cukes            # features/demo.feature:11
  Given I have 4 green cukes       # StuffSteps.iHaveNCukes(int,String)
  When I add a table               # StuffSteps.iAddATable(Table)
    | a | b |
    | 1 | 2 |
  And I add a string               # StuffSteps.iAddAString(String)
    """
    Hello
    World
    """
  Then I should have 4 green cukes # StuffSteps.iShouldHaveNCukes(int,String)
    java.lang.AssertionError: expected:<1> but was:<3> (NativeException)
```

I couldn't figure out a good way to fix it. There's a method called clearHooksAndStepDefinitions in AbstractProgrammingLanguage that looks like it's getting called per scenario, but it doesn't do anything with hooks.

**Comment by aslakhellesoy** [comment]
Bugfix: Hooks not being cleared per scenario. Closed by 4c82bc775277f53fa2146c5976d11c0f8e90215c.

**Comment by aslakhellesoy** [comment]
Not a great fix, but a fix nevertheless. The same bug might be in Scala. I'm adding a ticket for that.
